Sources/RemoteNotification.swift

```diff
@@ -33,28 +33,66 @@ public struct RemoteNotification: CustomStringConvertible, Equatable {
     /// The identifier for the UIUserNotificationCategory associated with the push notification. Derived from userInfo["category"]
     public let categoryIdentifier: String?
     /// Custom fields on the remote notification.
-    public let userInfo: [String : AnyObject]
+    public let userInfo: [String : Any]
     /// The dictionary representation of the remote notification.
-    public let remoteNotificationDictionary: [String : AnyObject]
+    public let remoteNotificationDictionary: [String : Any]
     
-    public init?(remoteNotification: [NSObject : AnyObject]?) {
-        guard let remoteNotification = remoteNotification as? [String : AnyObject] else {
+    // MARK: Equatable
+    
+    public static func ==(lhs: RemoteNotification, rhs: RemoteNotification) -> Bool {
+        if lhs.alert != nil || rhs.alert != nil {
+            guard let lhsAlert = lhs.alert, let rhsAlert = rhs.alert, lhsAlert == rhsAlert else {
+                return false
+            }
+        }
+        
+        if lhs.badge != nil || rhs.badge != nil {
+            guard let lhsBadge = lhs.badge, let rhsBadge = rhs.badge, lhsBadge == rhsBadge else {
+                return false
+            }
+        }
+        
+        if lhs.sound != nil || rhs.sound != nil {
+            guard let lhsSound = lhs.sound, let rhsSound = rhs.sound, lhsSound == rhsSound else {
+                return false
+            }
+        }
+        
+        guard lhs.contentAvailable == rhs.contentAvailable else {
+            return false
+        }
+        
+        if lhs.categoryIdentifier != nil || rhs.categoryIdentifier != nil {
+            guard let lhsCategoryIdentifier = lhs.categoryIdentifier, let rhsCategoryIdentifier = rhs.categoryIdentifier, lhsCategoryIdentifier == rhsCategoryIdentifier else {
+                return false
+            }
+        }
+        
+        guard NSDictionary(dictionary: lhs.userInfo) == NSDictionary(dictionary: rhs.userInfo) else {
```

Comment:
yeah. This is a bit of a hack. In Objc these dicts are comparable... and they should be Swift as well. But the value may not be `AnyHashable` since it might be a dictionary. Would really appreciate suggestions here.